app/assets/javascripts/discourse/components/search-advanced-options.js.es6

```diff
@@ -74,107 +86,204 @@ export default Em.Component.extend({
-    this.findStatus(searchTerm);
-    this.findPostsCount(searchTerm);
-    this.findPostTime(searchTerm);
+    this.findSearchTerms();
+    this.setSearchedTermValue('searchedTerms.username', REGEXP_USERNAME_PREFIX);
+    this.setSearchedTermValueForCategory();
+    this.setSearchedTermValueForGroup();
+    this.setSearchedTermValueForBadge();
+    this.setSearchedTermValueForTags();
+    this.setSearchedTermValue('searchedTerms.in', REGEXP_IN_PREFIX, REGEXP_IN_MATCH);
+    this.setSearchedTermSpecialInValue('searchedTerms.special.in.likes', REGEXP_SPECIAL_IN_LIKES_MATCH);
+    this.setSearchedTermSpecialInValue('searchedTerms.special.in.private', REGEXP_SPECIAL_IN_PRIVATE_MATCH);
+    this.setSearchedTermSpecialInValue('searchedTerms.special.in.wiki', REGEXP_SPECIAL_IN_WIKI_MATCH);
+    this.setSearchedTermValue('searchedTerms.status', REGEXP_STATUS_PREFIX);
+    this.setSearchedTermValueForPostTime();
+    this.setSearchedTermValue('searchedTerms.posts_count', REGEXP_POST_COUNT_PREFIX);
   },
 
-  findSearchTerm(EXPRESSION, searchTerm) {
+  findSearchTerms() {
+    const searchTerm = this.get('searchTerm');
     if (!searchTerm)
-      return "";
+      return [];
 
-    const expressionPosition = searchTerm.search(EXPRESSION);
-    if (expressionPosition === -1)
-      return "";
+    let result = [];
+    const blocks = searchTerm.match(REGEXP_BLOCKS);
+    _.each(blocks, function(block) {
+      if (block.length === 0) return;
 
-    const remainingPhrases = searchTerm.substring(expressionPosition + 2);
-    let nextExpressionPosition = remainingPhrases.search(REGEXP_FILTER_PREFIXES);
-    if (nextExpressionPosition === -1)
-      nextExpressionPosition = remainingPhrases.length;
+      result.push(block);
+    });
 
-    return searchTerm.substring(expressionPosition, nextExpressionPosition + expressionPosition + 2).trim().split(' ')[0];
+    this.set('searchTermBlocks', result);
   },
 
-  findUsername(searchTerm) {
-    const match = this.findSearchTerm(REGEXP_USERNAME_PREFIX, searchTerm);
+  filterBlocks(regexPrefix) {
+    let result = [];
+    _.each(this.get('searchTermBlocks'), function(block) {
+      if (block.search(regexPrefix) !== -1)
+        result.push(block);
+    });
 
-    if (match.length !== 0) {
-      let userInput = match.replace(REGEXP_USERNAME_PREFIX, '');
+    return result;
+  },
+
+  setSearchedTermValue(key, replaceRegEx, matchRegEx = null) {
+    matchRegEx = matchRegEx || replaceRegEx;
+    const match = this.filterBlocks(matchRegEx);
 
-      if (userInput.length !== 0 && this.get('searchedTerms.username') !== userInput) {
-        this.set('searchedTerms.username', userInput);
+    if (match.length !== 0) {
+      const userInput = match[0].replace(replaceRegEx, '');
+      if (this.get(key) !== userInput) {
+        this.set(key, userInput);
       }
-    } else {
-      this.set('searchedTerms.username', null);
+    } else if(this.get(key) !== null) {
+      this.set(key, null);
```

Comment:
Can do... but you actually originally changed searchedTerms.username to be set to null...
https://github.com/discourse/discourse/commit/4c2c21ee2b7cbb9c9f2915f94774372634c875e7

Even though they all default to '' in the _init.